bundles/admin/admin-announcements/publisher/AnnouncementsTool.js

```diff
@@ -0,0 +1,357 @@
+                    '<div>' +
+                        '<input type="text" name="publisher-announcements" disabled />' +
+                        '<button/>' +
+                    '</div>' +
+                '</div>'),
+            announcementsPopup: jQuery(
+                '<div>' +
+                    '<div id="publisher-announcements-inputs">' +
+                        '<h4>' + this.localization.tool.announcementsName + '</h4><h4>' + this.localization.tool.announcementsTime + '</h4>' +
+                        '<div class="ann-column" id="ann-title"></div>' +
+                        '<div class="ann-column" id="ann-time"/></div>' +
+                    '</div>' +
+                '</div>'),
+            inputCheckbox: jQuery('<div><input type="checkbox" /><label></label></div>')
+        };
+
+        this.eventHandlers = {
+            'Publisher2.ToolEnabledChangedEvent': function (event) {
+                var me = this;
+                var tool = event.getTool();
+                if (tool.getTool().id === me.getTool().id && tool.isStarted() && this.selectedAnnouncements) {
+                    this.getPlugin().updateAnnouncements(this.selectedAnnouncements);
+                }
+            }
+        };
+    }, {
+
+        init: function (data) {
+            const me = this;
+            me.data = data;
+            me.selectedAnnouncements = [];
+            me.annTitles = [];
+
+            const service = me.sandbox.getService('Oskari.framework.announcements.service.AnnouncementsService');
+
+            if (data.configuration && data.configuration.announcements && data.configuration.announcements.conf && data.configuration.announcements.conf.plugin) {
+                const myId = me.getTool().id;
+                const enabled = data.configuration.announcements.conf.plugin.id === myId;
+                me.setEnabled(enabled);
+            }
+
+            for (var p in me.eventHandlers) {
+                if (me.eventHandlers.hasOwnProperty(p)) {
+                    me.__sandbox.registerForEventByName(me, p);
+                }
+            }
+
+            service.fetchAdminAnnouncements((announcements) => {
+                me.announcements = announcements;
+                const toolPluginAnnouncementsConf = me._getToolPluginAnnouncementsConf();
+                if (toolPluginAnnouncementsConf !== null) {
+                    me.getPlugin().updateAnnouncements(toolPluginAnnouncementsConf.config.announcements);
+                    toolPluginAnnouncementsConf.config.announcements.forEach(announcement => {
+                        const filteredAnnouncement = me.announcements.filter(ann => ann.id === announcement);
+                        if (me.isAnnouncementValid(filteredAnnouncement[0])) {
+                            me.annTitles.push(filteredAnnouncement[0].title);
+                            me.selectedAnnouncements.push(filteredAnnouncement[0]);
```

Comment:
Is there a reason why the same data is in two different variables? I can see why there would be an "all announcements" and "selected announcements" that is a subset of all. But you have me.selectedAnnouncements + me.annTitles that is titles for the selecteded announcements and  me.announcements for the full data set.